src/App.js

```diff
@@ -1,66 +1,82 @@
 import './App.css'
+import 'antd/dist/antd.css'
 import React, { useEffect } from 'react'
 import {
   BrowserRouter as Router,
   Switch,
   Route,
-  Redirect,
   useRouteMatch,
+  Redirect,
 } from 'react-router-dom'
-import { SignIn } from './SignIn'
-import { ToDoList } from './ToDoList'
-import 'antd/dist/antd.css'
-import { useDispatch, useSelector } from 'react-redux'
 import { Button } from 'antd'
+import { useDispatch, useSelector } from 'react-redux'
 import { clearTodoList, logout } from './redux/actions'
+import { SignIn } from './SignIn'
+import { ToDoList } from './ToDoList'
 import { SignUp } from './SignUp'
 
 function App() {
   const user = useSelector((state) => state.userInfo.userInfo.uid)
 
   useEffect(() => {
-    console.log(user)
+    console.log('user', user)
   }, [user])
 
   return (
-    <Router>
+    <Router history={history}>
       <Switch>
-        <Route exact path='/'>
-          {user ? <Redirect to='/app' /> : <Redirect to='/signin' />}
-        </Route>
-        <Route path='/app'>
-          {user ? <AppRouter /> : <Redirect to='/signin' />}
-        </Route>
-        <Route path='/signin'>
-          {user ? <Redirect to='/app' /> : <SignIn />}
-        </Route>
-        <Route path='/signup'>
-          {user ? <Redirect to='/app' /> : <SignUp />}
-        </Route>
+        <Route
+          exact
+          path='/'
+          render={() =>
+            user ? <Redirect to='/app' /> : <Redirect to='/auth' />
```

Comment:
maybe it will be cleaner <Redirect to={user ? PATH_ROUTES.APP : PATH_ROUTES.AUTH} /> ?

Reply:
fixed